### ISSUE TITLE
Seeding initial data into in-memory database

### ISSUE THREAD

**Issue by KenxinKun** [fspcons/ports-service#4]
In reviewing the PR, it feels that the loading of JSON ports never actually gets written into the (in-memory) database? Instead it looks like for every create/update request, the entire JSON file is read again.

Was this the intended behaviour? How would a different solution address the performance issue of reading the file into memory each time?

**Comment by fspcons** [comment]
From the instructions file:

> - The file is of unknown size, it can contain several millions of records, you will not be able to read the entire file at once.
> - The service has limited resources available (e.g. 200MB ram).

As stated in those bullet points I feared I wouldn't be able to load a file of unknown size entirely to the memory, hence the read the file every time I received a request. 

Also on the limited resources note, my intentions there were to read the file by chunks coupled with reusing memory via sync Pools to reduce pressure on GC. 

I also stated in the comments, assuming the keys for the Ports Objects were written in alphabetical order (or any other logical order) on that JSON seed file, I'd recommend data partitioning between multiple files and/or using a binary search of sorts to improve the performance on the search for records on the file. 

> Was this the intended behaviour? How would a different solution address the performance issue of reading the file into memory each time?

Given the instructions, I'd say yes, that was the intended behavior. In a scenario with enough resources for it, I'd either load the entire file into the memory database (using the same "by chunks" with sync pool strategy) or simply load the file to an appropriate RDBS or NoSQL DB on the application start by an ensureSeed function.
